_posts/2020-09-06-how-the-system-tests-works.md

```diff
@@ -0,0 +1,493 @@
+Clusters could be Kubernetes, Minikube and so on.
+Lastly, in this module you can find [Command Client](https://github.com/strimzi/strimzi-kafka-operator/tree/master/test/src/main/java/io/strimzi/test/k8s/cmdClient),
+which servers as executor and can invoke various requests such as `kubectl annotate...`, `kubectl exec...` and so on.
+
+### crd-generator + api module
+
+Resources previously described in Resource section are used in crd-generator and api modules.
+Every time, when the code of Model changes you have to build the whole modules to generate new schema for resources. 
+For instance, imagine that you are using `Kafka` resource and you need to create Kafka with the `Cruise Control` support but you have old schemas, which does not support it.
+You will need to build these modules and generate schema, which will support it and be able to create Kafka with Cruise Control. 
+These modules provide the fluent way of creating Strimzi custom resource instances such as `Kafka`, `KafkaConnect`, `KafkaConnectS2I`, `KafkaMirrorMaker` and many other.
+
+## <a id="createtest">How to create a system test</a>
+
+At this point you have the necessary knowledge to create a simple test. There are also some additional steps, which need to be performed to set up your testing environment.
+
+### Precursors
+
+1. Fork the `strimzi-kafka-operator` repository - https://github.com/strimzi/strimzi-kafka-operator
+2. Build the whole project in root directory - `mvn clean install -DskipTests` or you can use `mvn -am clean install -DskipTests -f systemtest/pom.xml`
+3. Setup your Kubernetes cluster
+4. Login to your Kubernetes cluster
+5. Try to start some of our tests inside `system test` module for instance - `io.strimzi.systemtest.bridge.HttpBridgeST.testSendMessagesTlsAuthenticated()` 
+(test takes around 5 minutes)
+
+### Writing new test case
+
+We are going to create a very simple test using aforementioned classes like `Resources`, `Utils`, `Kubernetes client` and so on.
+
+Description of the test case: 
+
+1. Verify that Kafka was successfully deployed
+2. Clients are able to send and receive messages to Kafka cluster
+ 
+As you can see, we have two main things to take care of. First is `verify that Kafka was correctly deployed` and the second one is `clients can send messages to and receive messages from Kafka cluster`.
+
+Test scenario could look like this:
+
+1. Deploy Strimzi Operator <- setup + exercise
+2. Deploy `Kafka` cluster with 3 kafka nodes and 3 zookeeper nodes <- setup + exercise
+3. Verify that `Kafka` is ready <- verify
+4. Create instance of `Kafka clients` <- setup + exercise
+5. Send messages to `Kafka` cluster <- exercise
+6. Receive messages from `Kafka` cluster <- exercise
+7. Verify that we have sent and received expected count of messages <-verify
+8. Clean up all resources (this phase is implicitly done in AbstractST) <- teardown
```

Comment:
The `<-` are quite hard to read. I think you should try to rephrase the points to avoid it.